alf/optimizers/traj_optimizers.py

```diff
@@ -85,5 +87,6 @@ def obtain_solution(self, time_step: TimeStep, state):
         costs = self.cost_function(time_step, state, solutions)
         min_ind = torch.argmin(costs, dim=-1).long()
         # solutions [B, pop_size, sol_dim] -> [B, 1, sol_dim] -> [B, sol_dim]
-        solution = solutions.index_select(1, min_ind).squeeze(1)
+        solution = tensor_utils.batched_index_select(solutions, 1,
```

Comment:
solution = solutions[(torch.arange(batch_size), min_ind)]

Reply:
Replaced with this simpler method.